### PR TITLE
LPS-184588 [Marketplace] Implement Commerce Order Type through Site Initializer

### PR DESCRIPTION
**Pull request by joanagrego**
Related to --> [LPS-184582](https://issues.liferay.com/browse/LPS-184582)

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/build.gradle [hunk 1]
```diff
@@ -20,6 +20,7 @@ dependencies {
 	compileOnly project(":apps:commerce:commerce-product-api")
 	compileOnly project(":apps:commerce:commerce-price-list-api")
 	compileOnly project(":apps:commerce:headless:headless-commerce:headless-commerce-admin-catalog-api")
+	compileOnly project(":apps:commerce:headless:headless-commerce:headless-commerce-admin-order-api")
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/build.gradle:23]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-1/src/main/resources/site-initializer/commerce-order-type.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[
+	{
+		"active": false,
+		"description": {
+			"en_US": "TEST DESC 1"
+		},
+		"displayOrder": 1,
+		"externalReferenceCode": "TEST-ORDER-TYPE-1",
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-1/src/main/resources/site-initializer/commerce-order-type.json:8]: remove -

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/CommerceSiteInitializer.java [hunk 3]
```diff
@@ -127,6 +131,8 @@ public void addCPDefinitions(
 			servletContext);
 
 		_addOrUpdateCPOptionCategories(serviceContext, servletContext);
+
+		_addOrUpdateCommerceOrderType(serviceContext, servletContext);
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/CommerceSiteInitializer.java:135]: sort and remove blank line

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/CommerceSiteInitializer.java [hunk 4]
```diff
@@ -731,6 +737,47 @@ private Channel _addOrUpdateCommerceChannel(
...
+		OrderTypeResource orderTypeResource = builder.user(
+			serviceContext.fetchUser()
+		).build();
+
+		JSONArray jsonArray = _jsonFactory.createJSONArray(json);
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			OrderType orderType = OrderType.toDTO(
+				String.valueOf(jsonArray.getJSONObject(i)));
+
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/CommerceSiteInitializer.java:763]: prevent NPE

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/build.gradle (hunk 1 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/build.gradle:23]: Done

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-1/src/main/resources/site-initializer/commerce-order-type.json (hunk 2 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test-bundle-1/src/main/resources/site-initializer/commerce-order-type.json]: Done.

**Comment by joanagrego** [comment]
ci:test:sf

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 5]
```diff
@@ -722,6 +724,74 @@ private void _assertCommerceNotificationTemplate(
...
+		).build();
+
+		OrderType orderType =
+			orderTypeResource.getOrderTypeByExternalReferenceCode(
+				"TESTORDERTYPE1");
+
+		Map<String, String> orderTypeName = orderType.getName();
+
+		Assert.assertNotNull(orderType);
+
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:741]: do you really need this blank line?

Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:741]: Yes! Format Source.

**Comment by joanagrego** [comment]
ci:test:sf

**Review by nNilton — APPROVED** [review]

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:reevaluate:1277401_8220

**Comment by joseabelenda** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:forward

**Pull request closed without merging**